packages/driver/src/options.ts

```diff
@@ -150,33 +160,56 @@ export class Session {
     this.globals = globals;
   }
 
+  private _clone(mergeOptions: SessionOptions) {
+    const session = new Session({ ...this, ...mergeOptions });
+    session.annotations = this.annotations;
+    return session;
+  }
+
   withModuleAliases({
     module,
     ...aliases
   }: {
     [name: string]: string;
   }): Session {
-    return new Session({
-      ...this,
+    return this._clone({
       module: module ?? this.module,
       moduleAliases: { ...this.moduleAliases, ...aliases },
     });
   }
 
   withConfig(config: { [name: string]: any }): Session {
-    return new Session({
-      ...this,
+    return this._clone({
       config: { ...this.config, ...config },
     });
   }
 
   withGlobals(globals: { [name: string]: any }): Session {
-    return new Session({
-      ...this,
+    return this._clone({
       globals: { ...this.globals, ...globals },
     });
   }
 
+  withTag(tag: string | null): Session {
```

Comment:
I'm not sure exactly what api we want here? It looks like the rust client has a `withTag` api (also with reserved tag names) (https://github.com/edgedb/edgedb-rust/pull/364/files#diff-608add6cc9859de62fac8b605b75549b985354b28530c7c1b07180ccd86fd729R574), but the python implementation just exposes the underlying annotations? (https://github.com/edgedb/edgedb-python/pull/552/files#diff-ac82d0a568b6ac577eb791f28078350639f6ffd9b904254a17557bb7dff1fb62R416)

Reply:
Ah maybe I should just stay with `with_tag()` in Python too (so that the same tag limitation can be applied). cc @1st1 ?

Reply:
`withQueryTag()` it is then.